### PR TITLE
[SPARK-47513][TESTS] Regenerate benchmark results

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to regenerate benchmark results for Apache Spark 4.0.0.

### Why are the changes needed?

Recently, we changed many dependencies like Hadoop 3.4.0 and ORC 2.0.0. To keep track the micro-benchmark result, w had better update them.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual review.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: sql/hive/benchmarks/OrcReadBenchmark-jdk21-results.txt [hunk 1]
```diff
@@ -2,221 +2,221 @@
...
+OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 6.5.0-1016-azure
 AMD EPYC 7763 64-Core Processor
 Nested Struct Scan with 10 Elements, 30 Fields:  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------------
-Hive built-in ORC                                        9739           9740           2          0.1        9287.4       1.0X
-Native ORC MR                                            5553           5613          84          0.2        5296.0       1.8X
-Native ORC Vectorized                                    2090           2108          26          0.5        1993.1       4.7X
+Hive built-in ORC                                        4417           4430          19          0.2        4212.1       1.0X
+Native ORC MR                                            5039           5069          42          0.2        4805.9       0.9X
+Native ORC Vectorized                                    1726           1737          16          0.6        1645.8       2.6X
```
Comment (dongjoon-hyun) [inline sql/hive/benchmarks/OrcReadBenchmark-jdk21-results.txt:220]: The new ratio shows that all performance improved in nested struct scan. `Hive built-in ORC` is the most improved one.

Comment (yaooqinn) [reply sql/hive/benchmarks/OrcReadBenchmark-jdk21-results.txt:220]: Impressive! 

Will the related PR be marked release highlight? And does it mean that SPARK-44677 should be planned later?

**Review by yaooqinn — APPROVED** [review]
LGTM

File: sql/hive/benchmarks/OrcReadBenchmark-jdk21-results.txt (hunk 1 above)
Comment (dongjoon-hyun) [reply sql/hive/benchmarks/OrcReadBenchmark-jdk21-results.txt:220]: It's too early to say any improvement. I hope there is no regression until Apache Spark 4.0.0 release. :) 

For SPARK-44677, I believe SPARK-44444 would be the precondition. For now, we didn't make any official agreement yet.

File: sql/core/benchmarks/BloomFilterBenchmark-jdk21-results.txt [hunk 2]
```diff
@@ -2,191 +2,191 @@
...
 
-OpenJDK 64-Bit Server VM 21.0.1+12-LTS on Linux 5.15.0-1053-azure
+OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 6.5.0-1016-azure
 AMD EPYC 7763 64-Core Processor
 Read a row from 100M rows:                Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-Without bloom filter, blocksize: 6291456           1247           1252           6         80.2          12.5       1.0X
-With bloom filter, blocksize: 6291456               865            879          21        115.6           8.7       1.4X
+Without bloom filter, blocksize: 6291456            805            822          20        124.3           8.0       1.0X
+With bloom filter, blocksize: 6291456               527            582          72        189.9           5.3       1.5X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/BloomFilterBenchmark-jdk21-results.txt:46]: ORC Read with BloomFilter is improved.

File: sql/hive/benchmarks/OrcReadBenchmark-jdk21-results.txt (hunk 1 above)
Comment (yaooqinn) [reply sql/hive/benchmarks/OrcReadBenchmark-jdk21-results.txt:220]: Thanks for the information. @dongjoon-hyun

File: sql/core/benchmarks/BuiltInDataSourceWriteBenchmark-jdk21-results.txt [hunk 3]
```diff
@@ -2,69 +2,69 @@
...
 
 
 ================================================================================================
 ORC writer benchmark
 ================================================================================================
 
-OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 5.15.0-1053-azure
+OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 6.5.0-1016-azure
 AMD EPYC 7763 64-Core Processor
 ORC writer benchmark:                     Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/BuiltInDataSourceWriteBenchmark-jdk21-results.txt:32]: ORC write is improved.

File: sql/core/benchmarks/FilterPushdownBenchmark-jdk21-results.txt [hunk 4]
```diff
@@ -2,733 +2,733 @@
...
 --------------------------------------------------------------------------------------------------------------------------
-Parquet Vectorized                                  12805          12872          92          1.2         814.1       1.0X
-Parquet Vectorized (Pushdown)                       12833          12851          14          1.2         815.9       1.0X
-Native ORC Vectorized                               13594          13615          16          1.2         864.3       0.9X
-Native ORC Vectorized (Pushdown)                    13626          13637          11          1.2         866.3       0.9X
+Parquet Vectorized                                  12335          12564         207          1.3         784.2       1.0X
+Parquet Vectorized (Pushdown)                       12561          12587          20          1.3         798.6       1.0X
+Native ORC Vectorized                               11278          11295          12          1.4         717.0       1.1X
+Native ORC Vectorized (Pushdown)                    11398          11468          98          1.4         724.7       1.1X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/FilterPushdownBenchmark-jdk21-results.txt:57]: Although the ratio is reversed slightly, the different is too small.

File: sql/core/benchmarks/FilterPushdownBenchmark-jdk21-results.txt [hunk 5]
```diff
@@ -2,733 +2,733 @@
...
 ------------------------------------------------------------------------------------------------------------------------
-Parquet Vectorized                                 6968           6990          14          2.3         443.0       1.0X
-Parquet Vectorized (Pushdown)                      1456           1487          42         10.8          92.6       4.8X
-Native ORC Vectorized                              7284           7290           5          2.2         463.1       1.0X
-Native ORC Vectorized (Pushdown)                   1488           1497           8         10.6          94.6       4.7X
+Parquet Vectorized                                 6624           6642          20          2.4         421.1       1.0X
+Parquet Vectorized (Pushdown)                      1438           1450          11         10.9          91.4       4.6X
+Native ORC Vectorized                              5086           5101          17          3.1         323.4       1.3X
+Native ORC Vectorized (Pushdown)                   1293           1303           8         12.2          82.2       5.1X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/FilterPushdownBenchmark-jdk21-results.txt:120]: ditto. Although the ratio is reversed, the different is small.

File: sql/core/benchmarks/FilterPushdownBenchmark-jdk21-results.txt [hunk 6]
```diff
@@ -2,733 +2,733 @@
...
 ------------------------------------------------------------------------------------------------------------------------
-Parquet Vectorized                                 9072           9082          11          1.7         576.8       1.0X
-Parquet Vectorized (Pushdown)                      6042           6065          21          2.6         384.1       1.5X
-Native ORC Vectorized                              9554           9580          17          1.6         607.4       0.9X
-Native ORC Vectorized (Pushdown)                   6387           6400           9          2.5         406.1       1.4X
+Parquet Vectorized                                 8735           8748           8          1.8         555.3       1.0X
+Parquet Vectorized (Pushdown)                      5861           5871          12          2.7         372.7       1.5X
+Native ORC Vectorized                              7281           7331          29          2.2         462.9       1.2X
+Native ORC Vectorized (Pushdown)                   5242           5250          12          3.0         333.3       1.7X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/FilterPushdownBenchmark-jdk21-results.txt:129]: Th ratio is slightly reversed.

File: sql/core/benchmarks/FilterPushdownBenchmark-jdk21-results.txt [hunk 7]
```diff
@@ -2,733 +2,733 @@
...
-Parquet Vectorized                                                 6568           6590          35          2.4         417.6       1.0X
-Parquet Vectorized (Pushdown)                                       871            875           3         18.1          55.4       7.5X
-Native ORC Vectorized                                              6956           6967          11          2.3         442.3       0.9X
-Native ORC Vectorized (Pushdown)                                    616            624           7         25.5          39.2      10.7X
+Parquet Vectorized                                                 6275           6286           8          2.5         398.9       1.0X
+Parquet Vectorized (Pushdown)                                       839            843           3         18.8          53.3       7.5X
+Native ORC Vectorized                                              4747           4763          17          3.3         301.8       1.3X
+Native ORC Vectorized (Pushdown)                                    474            479           5         33.2          30.1      13.2X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/FilterPushdownBenchmark-jdk21-results.txt:528]: InFilters seems to be improved.

File: sql/core/benchmarks/JsonBenchmark-results.txt [hunk 8]
```diff
@@ -3,128 +3,128 @@ Benchmark for performance of JSON parsing
...
 Preparing data for benchmarking ...
-OpenJDK 64-Bit Server VM 17.0.9+9-LTS on Linux 5.15.0-1053-azure
+OpenJDK 64-Bit Server VM 17.0.10+7-LTS on Linux 6.5.0-1016-azure
 AMD EPYC 7763 64-Core Processor
 count a wide column:                      Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-No encoding                                        4483           4590         150          0.2        4483.4       1.0X
-UTF-8 is set                                       4020           4051          38          0.2        4019.7       1.1X
+No encoding                                        3185           3258         122          0.3        3185.0       1.0X
+UTF-8 is set                                       4058           4093          42          0.2        4058.1       0.8X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/JsonBenchmark-results.txt:27]: The ratio is reversed noticeably. This seems to be a regression on UTF-8.

File: sql/core/benchmarks/MakeDateTimeBenchmark-jdk21-results.txt [hunk 9]
```diff
@@ -1,22 +1,22 @@
-OpenJDK 64-Bit Server VM 21.0.1+12-LTS on Linux 5.15.0-1053-azure
+OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 6.5.0-1016-azure
 AMD EPYC 7763 64-Core Processor
 make_date():                              Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-prepare make_date()                                2004           2018          13         49.9          20.0       1.0X
-make_date(2019, 9, 16)                             2694           2737          52         37.1          26.9       0.7X
-make_date(*, *, *)                                 3881           3906          37         25.8          38.8       0.5X
+prepare make_date()                                1981           2010          32         50.5          19.8       1.0X
+make_date(2019, 9, 16)                             1833           1849          14         54.5          18.3       1.1X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/MakeDateTimeBenchmark-jdk21-results.txt:6]: `make_date` seems to be improved.

File: sql/core/benchmarks/MetricsAggregationBenchmark-results.txt [hunk 10]
```diff
@@ -1,12 +1,12 @@
-OpenJDK 64-Bit Server VM 17.0.9+9-LTS on Linux 5.15.0-1053-azure
+OpenJDK 64-Bit Server VM 17.0.10+7-LTS on Linux 6.5.0-1016-azure
 AMD EPYC 7763 64-Core Processor
 metrics aggregation (50 metrics, 100000 tasks per stage):  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ----------------------------------------------------------------------------------------------------------------------------------------
-1 stage(s)                                                          650            668          16          0.0   650409908.0       1.0X
-2 stage(s)                                                          931           1010         111          0.0   931458679.0       0.7X
-3 stage(s)                                                         1916           2000         119          0.0  1915675055.0       0.3X
+1 stage(s)                                                          614            653          49          0.0   613999403.0       1.0X
+2 stage(s)                                                         1306           1324          25          0.0  1306014655.0       0.5X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/MetricsAggregationBenchmark-results.txt:6]: This change is consistent with Java 21 result.

File: sql/core/benchmarks/TPCDSQueryBenchmark-jdk21-results.txt [hunk 11]
```diff
@@ -1,810 +1,810 @@
...
-q22a-v2.7                                          1853           1888          49          6.4         156.6       1.0X
+q22a-v2.7                                          1930           1938          12          6.1         163.0       1.0X
 
-OpenJDK 64-Bit Server VM 21.0.1+12-LTS on Linux 5.15.0-1053-azure
+OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 6.5.0-1016-azure
 AMD EPYC 7763 64-Core Processor
 TPCDS:                                    Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-q24-v2.7                                            212            254          35         15.7          63.5       1.0X
+q24-v2.7                                            101            230          75         33.2          30.2       1.0X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/TPCDSQueryBenchmark-jdk21-results.txt:689]: The time is too small to say something.

**Comment by dongjoon-hyun** [comment]
There is no noticeably outstanding regression. Let me merge this as a new baseline.

**Pull request closed without merging**